src/rocksdb_options.rs

```diff
@@ -158,7 +158,7 @@ impl UnsafeSnap {
 
 pub struct ReadOptions {
     inner: *mut DBReadOptions,
-    upper_bound: Vec<u8>,
+    upper_bound: Option<Vec<u8>>,
```

Comment:
`Vec` points a continuous memory on heap already, I think it's ridiculous to use `Box` again, which will allocate again.